### PR TITLE
rework topics page to be more crate-based

### PR DESCRIPTION
**Pull request by whispersofthedawn**
This does two things:
- Update the ecosystem file to use a more crate-based approach to crates.io and docs.rs links.
- Update the UI of the topics page to use fontawesome icons as opposed to plain links.

image

### CODE REVIEW

File: templates/base.tera.html [hunk 1]
```diff
@@ -15,6 +15,9 @@
     <meta property="og:type" content="website" />
 
     <link rel="stylesheet" href="/theme.css">
+
+    <!-- icons -->
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css">
```
Comment (Legorooj) [inline templates/base.tera.html]: I'm not a big fan of FontAwesome due to their licensing in the past, would we be able to use https://primer.style/octicons/ instead?

File: templates/topic.tera.html [hunk 2]
```diff
@@ -8,25 +8,32 @@ <h1>{{ topic.name }}</h1>
...
+        {% if project.repo %}
+            <a title="repository" href="{{ project.repo }}"><i class="fa fa-code"></i></a>
+        {% endif %}
+    </h2>
 
     {% if project.description %}
         <p>{{ project.description }}</p>
     {% endif %}
     
+    {% if project.crates     > 1 %}
```
Comment (Legorooj) [inline templates/topic.tera.html]:
```suggestion
    {% if project.crates %}
```
This PR won't build on my PC without this change.

**Review by Legorooj — CHANGES_REQUESTED** [review]

File: templates/base.tera.html (hunk 1 above)
Comment (whispersofthedawn) [reply templates/base.tera.html]: Sure, I can migrate that.

File: templates/topic.tera.html (hunk 2 above)
Comment (whispersofthedawn) [reply templates/topic.tera.html]: https://github.com/arewelangyet/arewelangyet/pull/18/commits/d4cc854d89c2ff81be6a8fe1a9a83270d44d676e

File: templates/base.tera.html (hunk 1 above)
Comment (whispersofthedawn) [reply templates/base.tera.html]: https://github.com/arewelangyet/arewelangyet/pull/18/commits/d4cc854d89c2ff81be6a8fe1a9a83270d44d676e

**Review by Legorooj — APPROVED** [review]

**Pull request merged**